OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs

```diff
@@ -106,21 +106,7 @@ protected virtual bool ShouldFlee(Squad squad, Func<IReadOnlyCollection<Actor>,
 
 		protected static bool IsRearming(Actor a)
 		{
-			if (a.IsIdle)
-				return false;
-
-			var activity = a.CurrentActivity;
-			if (activity.GetType() == typeof(Resupply))
-				return true;
-
-			var next = activity.NextActivity;
-			if (next == null)
-				return false;
-
-			if (next.GetType() == typeof(Resupply))
-				return true;
-
-			return false;
+			return !a.IsIdle && a.CurrentActivity.ActivitiesImplementing<Resupply>().Any();
```

Comment:
it should be checking for ReturnToBase activity as well, though both of these activities will be replaced by the docking refactor

Reply:
Fixed